voluncheer/opportunityboard/views/search.py

```diff
@@ -65,10 +66,32 @@ def search(self):
             filtered_opportunity = filtered_opportunity.filter(subcategory=self.subcategory)
         if self.subsubcategory is not None:
             filtered_opportunity = filtered_opportunity.filter(subsubcategory=self.subsubcategory)
-
+        if self.duration == "0-2 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 0, 2)
+        if self.duration == "2-4 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 2, 4)
+        if self.duration == "4-8 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 4, 8)
+        if self.duration == "8+ Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 8, 24)
         return filtered_opportunity
 
 
+def filter_by_duration(queryset, min, max):
```

Comment:
Did you considered using just a maximum instead of a range? I'm wondering what caused you to go for a range rather than just a maximum.

Reply:
Looking over the acceptance criteria, I think your suggestion is closer to what was specified, so I changed the function accordingly. Assuming "Full-day" means 8 hours, but curious what Hanna thinks.